pkg/cmd/api/api.go

```diff
@@ -417,15 +419,26 @@ func fillPlaceholders(value string, opts *ApiOptions) (string, error) {
 			name = m[1 : len(m)-1]
 		}
 
+		overriddenValue, overridden := os.LookupEnv("GH_REPO")
+		overriddenOwner, overriddenRepo := "", ""
+		if overriddenValue != "" {
+			overriddenOwner = strings.Split(overriddenValue, "/")[0]
+			overriddenRepo = strings.Split(overriddenValue, "/")[1]
+		}
```

Comment:
This works but it duplicates a part of EnableRepoOverride functionality. Let's see if we can reuse it!

Our first priority should be to undo all current edits to `gh api` implementation. The command consumes `BaseRepo()` information passed from the "outside", and thus it should not have awareness of where BaseRepo comes from and whether or not it was overridden by GH_REPO.

After that, let's make sure that we pass an implementation of BaseRepo that is informed by GH_REPO. Other commands are able to use `EnableRepoOverride`, but that adds a `-R/--repo` flag to the command, which we don't want here for now. So let's use `OverrideBaseRepoFunc` instead.

Something like this could work when creating the Cobra command for `api`?
```go
PreRun: function(cmd *cobra.command, args []string) {
  f.BaseRepo = OverrideBaseRepoFunc(f, "")
},
```
It's not the most idiomatic but it would do the trick for now. If you go this route, don't worry about adding GH_REPO-specific tests to the `api` command, since that's not necessary.